src/app.js

```diff
@@ -0,0 +1,306 @@
+/*!
+GPII Application
+Copyright 2016 Steven Githens
+Copyright 2016-2017 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+"use strict";
+
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+var Menu = require("electron").Menu;
+var Tray = require("electron").Tray;
+var path = require("path");
+var request = require("request");
+
+/*
+ ** Component to manage the app.
+ */
+fluid.defaults("gpii.app", {
+    gradeNames: "fluid.modelComponent",
+    model: {
+        // This model has a "menu" item that is only relayed from gpii.app.menu,
+        // Don't uncomment the line below so that the initial menu can be populated
+        // by gpii.app.menu component using the initial model.keyedInUserToken.
+        // menu: null,
+        keyedInUserToken: null
+    },
+    // The list of the default snapsets shown on the initial task tray menu for key-in
+    snapsets: {
+        alice: "Alice",
+        davey: "Davey",
+        david: "David",
+        elaine: "Elaine",
+        elmer: "Elmer",
+        elod: "Elod",
+        livia: "Livia"
+    },
+    members: {
+        tray: {
+            expander: {
+                funcName: "gpii.app.makeTray",
+                args: ["{that}.options.icon"]
+            },
+            createOnEvent: "onGPIIReady"
+        }
+    },
+    components: {
+        menu: {
+            type: "gpii.app.menu",
+            createOnEvent: "onGPIIReady",
+            options: {
+                model: {
```

Comment:
All of this material should be factored out into a top-level addin grade named, say, gpii.app.menuInApp